### PR TITLE
fix: gRPC errors regarding group cols

### PR DESCRIPTION
**Pull request by crepererum**
- missing group col prev. produced an "internal error" but should be "invalid argument"
- duplicate group cols produced a panic but should also be "invalid argument"

### CODE REVIEW

**Comment by crepererum** [comment]
This needs another iteration, because some names are treated specially (e.g. `_field`).

**Comment by crepererum** [comment]
> This needs another iteration, because some names are treated specially (e.g. `_field`).

done.

File: predicate/src/rpc_predicate.rs [hunk 1]
```diff
@@ -57,6 +57,24 @@ pub const FIELD_COLUMN_NAME: &str = "_field";
...
 pub const VALUE_COLUMN_NAME: &str = "_value";
 
+/// Special group key for `read_group` requests.
+///
+/// Treat these specially and use value "" to mirror what TSM does
+/// see <https://github.com/influxdata/influxdb_iox/issues/2693#issuecomment-947695442>
+/// for more details.
+///
+/// See also [`GROUP_KEY_SPECIAL_STOP`].
+pub const GROUP_KEY_SPECIAL_START: &str = "_start";
```
Comment (alamb) [inline predicate/src/rpc_predicate.rs:67]: The comments say use value "" but then the constant is "_start" which I found confusing

File: influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs [hunk 2]
```diff
@@ -207,6 +207,90 @@ async fn test_read_group_periods() {
...
+            Step::Custom(Box::new(move |state: &mut StepTestState| {
+                async move {
+                    let grpc_connection = state
+                        .cluster()
+                        .querier()
+                        .querier_grpc_connection()
+                        .into_grpc_connection();
+                    let mut storage_client = StorageClient::new(grpc_connection);
+
+                    let group_keys = match variant {
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/querier/influxrpc/read_group.rs:258]: nice

**Review by alamb — APPROVED** [review]
Seems reasonable to me -- thank you @crepererum

File: predicate/src/rpc_predicate.rs (hunk 1 above)
Comment (crepererum) [reply predicate/src/rpc_predicate.rs:67]: improved.

**Pull request merged**